src/vue/src/main.js

```diff
@@ -175,11 +175,21 @@ new Vue({
             }
 
             if (this.$hasPermission('can_view_journals_of_own_groups', 'assignment', assignment.id)) {
-                if (!assignment.is_published && !assignment.is_group_assignment) { // Teacher not published route
-                    if (this.$hasPermission('can_edit_assignment', 'assignment', assignment.id)) {
-                        route.name = 'AssignmentEditor'
-                    }
-                } else { // Teacher published route
+                if (assignment.is_published) {
+                    // Published assignments, always go to assignment
+                    route.name = 'Assignment'
+                } else if (!this.$hasPermission('can_view_unpublished_assignment', 'assignment', assignment.id)) {
+                    // Unpublished assignments require the `can_view_unpublished_assignment` permission, else throw
+                    // error and redirect to homepage (redirect is done below)
+                    this.$toasted.error('You are not yet allowed to view this assignment.')
```

Comment:
```suggestion
                    this.$toasted.error('You are not allowed to view this assignment.')
```
Could be that this is never even going to be the case.